src/math_tools/gradient.hpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+#include "metric_tensor_evaluator.hpp"
+
+
+/**
+ * @brief A class which implements a gradient operator 
+ * @tparam Mapping A mapping.
```

Comment:
Maybe more specific?
```suggestion
 * @tparam Mapping A coordinate transformation mapping to the Cartesian coordinates.
```